File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Matemática/Big_Factorial.cpp
#include <iostream>

using namespace std;

int multiply(int x, int res[], int res_size){
    int carry = 0;
    for(int i = 0; i < res_size; i++){
        int prod = res[i] * x + carry;
        res[i] = prod % 10;
        carry = prod / 10;
    }
    while(carry != 0){
        res[res_size] = carry % 10;
        carry /= 10;
        res_size++;
    }
    return res_size;
}

void factorial(int n)
{
    int res[10000];
    res[0] = 1;
    int res_size = 1;
    for(int i = 2; i <= n; i++){
        res_size = multiply(i, res, res_size);
    }
    cout<<"Factorial del numero dado es:"<<endl;
    for(int i = res_size - 1; i >= 0; i--){
        cout<<res[i];
    }
    cout<<endl;
}

int main()
{
    factorial(100);
    return 0;
}

File: Codigos estudiantes por lenguaje/Java/Edwin Villarraga/matematicas/BigFactorial.java
public class Main {
    static void factorial(int n) {
        int res[] = new int[500];
        res[0] = 1;
        int res_size = 1;

        for (int x = 2; x <= n; x++) {
            res_size = multiply(x, res, res_size);
        }
        System.out.println("Factorial del numero dado es ");
        for (int i = res_size - 1; i >= 0; i--) {
            System.out.print(res[i]);
        }
    }
    static int multiply(int x, int res[], int res_size) {
        int carry = 0; // inicializar carry
        for (int i = 0; i < res_size; i++) {
            int prod = res[i] * x + carry;
            res[i] = prod % 10;
            carry = prod / 10;
        }
        while (carry != 0) {
            res[res_size] = carry % 10;
            carry = carry / 10;
            res_size++;
        }
        return res_size;
    }
    public static void main(String args[]) {
        factorial(100);
    }
}
